### PR TITLE
Add: dropdown to show all assistants used by a datasource and show modal on click

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Allow showing the modal of each used assistants via a dropdown.
Added a lookup endpoint (name => sId) as adding sId to the usage endpoint was much much more involved (and the query is not cheap).

**References:**
- https://github.com/dust-tt/tasks/issues/1602

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/spaces/UsedByButton.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <DropdownMenuTrigger asChild>
+        <Button
+          icon={RobotIcon}
+          variant="ghost"
+          isSelect
+          size="sm"
+          label={`${usage.count}`}
+        />
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="max-w-[300px]">
```
Comment (JulesBelveze) [inline front/components/spaces/UsedByButton.tsx:30]: We need to handle when there's no agent. Probably disable the `Button` and set `isSelect={false}` in this case?

File: front/components/spaces/UsedByButton.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        <ScrollArea className="border-1 max-h-[300px]">
+          {usage.agentNames.map((name) => (
+            <DropdownMenuItem
```
Comment (JulesBelveze) [inline front/components/spaces/UsedByButton.tsx:33]: Let's display the avatar image as well?

File: front/components/spaces/UsedByButton.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <Button
+          icon={RobotIcon}
+          variant="ghost"
+          isSelect
+          size="sm"
+          label={`${usage.count}`}
+        />
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="max-w-[300px]">
+        <ScrollArea className="border-1 max-h-[300px]">
```
Comment (JulesBelveze) [inline front/components/spaces/UsedByButton.tsx:31]: This won't work, `ScrollArea` needs a fixed size

File: front/components/spaces/SpaceResourcesList.tsx [hunk 4]
```diff
@@ -135,13 +139,11 @@ const getTableColumns = ({
...
-            icon={RobotIcon}
-            title={`Used by ${info.row.original.dataSourceView.usage.agentNames.join(
-              ", "
-            )}`}
-          >
-            {info.row.original.dataSourceView.usage.count}
+          <DataTable.CellContent>
+            <UsedByButton
+              usage={info.row.original.dataSourceView.usage}
+              onItemClick={setAssistantName}
```
Comment (JulesBelveze) [inline front/components/spaces/SpaceResourcesList.tsx:145]: Could we use the router instead? Like done in the `ConversationLayout`

File: front/components/spaces/UsedByButton.tsx (hunk 1 above)
Comment (Fraggle) [reply front/components/spaces/UsedByButton.tsx:30]: Good point !

File: front/components/spaces/SpaceResourcesList.tsx (hunk 4 above)
Comment (JulesBelveze) [reply front/components/spaces/SpaceResourcesList.tsx:145]: 👌🏼

File: front/components/spaces/UsedByButton.tsx (hunk 3 above)
Comment (Fraggle) [reply front/components/spaces/UsedByButton.tsx:31]: Thanks, fixed.

**Review by JulesBelveze — APPROVED** [review]
LGTM

**Pull request merged**